chain/chain/src/chain.rs

```diff
@@ -2542,8 +2542,46 @@ impl Chain {
         &mut self,
         block_hash: &CryptoHash,
         shard_id: ShardId,
-    ) -> Result<Option<&CryptoHash>, Error> {
-        self.store.get_next_block_hash_with_new_chunk(block_hash, shard_id)
+    ) -> Result<Option<(CryptoHash, ShardId)>, Error> {
```

Comment:
Could we update the comment for this function to include what it does when the number of shards changes?